layouts/shortcodes/flexcolumns.html

```diff
@@ -0,0 +1,34 @@
+{{- if .IsNamedParams -}}
+  {{- $.Scratch.Set "ratio" (.Get "ratio") -}}
+  {{- $.Scratch.Set "mobile" (.Get "flattenInMobileView") -}}
+{{- else -}}
+  {{- $.Scratch.Set "ratio" (.Get 0) -}}
+  {{- $.Scratch.Set "mobile" (.Get 1) -}}
+{{- end -}}
+
+{{- if eq ($.Scratch.Get "mobile") "true" -}}
+  {{- $.Scratch.Set "mobile" "flatten-in-mobile" -}}
+{{- end -}}
+
+{{- if $.Scratch.Get "ratio" -}}
+  {{- $.Scratch.Set "ratio" (split ($.Scratch.Get "ratio") ":") -}}
+{{- end -}}
+
+<div class="book-columns flex flex-wrap {{ $.Scratch.Get "mobile" }}">
+{{- if $.Scratch.Get "ratio" -}}
+  {{- range $i, $val := split .Inner "<--->" -}}
+    {{- $ratioSplit := $.Scratch.Get "ratio" -}}
+    {{- if gt (len $ratioSplit) $i}}
+      <div style="flex-grow:{{ index $ratioSplit $i }};" class="flex-even markdown-inner">
+        {{- $val | $.Page.RenderString -}}
+      </div>
+    {{- end -}}
```

Comment:
Can you help me understand this block?

If I understand correctly, the ratio is split into its pieces beforehand (i.e. the ratio `1:2:3` becomes the slice `[1, 2, 3]`) and `$i` is the current index of the markdown blocks split by `<--->`. So this conditional checks to see if `$i` is greater than the length of the slice of ratio values and, if so - only if so? - adds a `div` containing the rendered content with the `flex-grow` CSS attribute set to the content's index in the slice of ratio values.

What happens when the declared ratio is `1:2:3` and there's only two sections in the markdown? When there's four sections? My brain seems to think the first item in the split markdown (index `0`?) will always be less than the length of the slice of ratio values if they're non-zero - but your code is working, which means I'm definitely misunderstanding. 😭

Reply:
> So this conditional checks to see if $i is greater than the length of the slice of ratio values

gt works the other way: the slice length is greater than $i.

Reply:
Ahhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh

I _get_ why the functions are the way they are but I would kill for it to have logical operators implemented so it could be `if (len $ratio) > $i` 😭